packages/notifi-dapp-example/src/components/DiscordInput.tsx

```diff
@@ -9,7 +17,50 @@ export type DiscordInputProps = Readonly<{
 }>;
 
 export const DiscordInput: React.FC<DiscordInputProps> = ({ disabled }) => {
-  const { useDiscord } = useNotifiSubscriptionContext();
+  const { formState } = useNotifiForm();
+  const { setIsGlobalLoading, setGlobalError } = useGlobalStateContext();
+  const { useDiscord, setUseDiscord, render } = useNotifiSubscriptionContext();
+  const { frontendClient } = useNotifiClientContext();
+  const { renewTargetGroups } = useNotifiTargets();
+
+  const { phoneNumber, telegram: telegramId, email } = formState;
+
+  const targetGroup: TargetGroupData = useMemo(
```

Comment:
Per our discussion before, instead of repeating all this, shall we create a custom hook to handle this